tests/build.proj

```diff
@@ -51,8 +51,8 @@
   </Target>
 
   <Target Name="RestorePackage">
-    <Exec Condition="'$(RunningOnCore)' == 'false'" Command="$(DotnetRestoreCommand) $(RestoreProj) $(PackageVersionArg)" StandardOutputImportance="Low" />
-    <Exec Condition="'$(RunningOnCore)' == 'true'"  Command="$(DotnetRestoreCommand) -r $(__DistroRid) $(RestoreProj) $(PackageVersionArg)" StandardOutputImportance="Low" />
+    <Exec Condition="'$(__DistroRid)' == ''" Command="$(DotnetRestoreCommand) $(RestoreProj) $(PackageVersionArg)" StandardOutputImportance="Low" />
+    <Exec Condition="'$(__DistroRid)' != ''" Command="$(DotnetRestoreCommand) -r $(__DistroRid) $(RestoreProj) $(PackageVersionArg)" StandardOutputImportance="Low" />
```

Comment:
What exactly is the pain point? Does it fail? Does it _not_ properly fully restore? I am more than willing to add logging, but I would like to understand pain points because there may be a more general fix.

Reply:
The problem is that there are multiple already attempts at `fixing` dependencies in particular for performance tests where majority of them failed from different reasons.  See #16647 #9281 #16126. Discussion related to the problem was in some comments of the https://github.com/dotnet/corefx/issues/31536#issuecomment-409635103 issue.

In general build-test configuration causes download of roughly 2.5 GB of assemblies out of which only about 500 - 600 are necessary. There is a possibility of reducing that amount by careful management of  dependencies. This will lead to deterministic build which should be made against built product and chosen CoreFX or netstandard assemblies. Detailed information on dependency graph resolution would make any feature test dependencies updates easier.

Part of the above goal can be achieved by introducing new features into NuGet which would allow to override resolution of inbox assemblies with latest used version.

Solution which will not impact standard builds could be based on verbosity parameter passed to build-test.(cmd|sh) which would be used in the above code to set logging verbosity for restore and visibility of command line messages in console output by changing `StandardOutputImportance`

Reply:
That is indeed a valuable goal here. In this case, however it is unrelated. Providing a new knob to configure logging for the update isn't the purpose of this PR it is for a long standing work item related to consolidating usage of `dotnet`'s MSBuild version.